src/fsharp/FSharp.Build/FSharpEmbedResXSource.fs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.FSharp.Build
+
+open System
+open System.Collections
+open System.Globalization
+open System.IO
+open System.Linq
+open System.Text
+open System.Xml.Linq
+open Microsoft.Build.Framework
+open Microsoft.Build.Utilities
+
+type FSharpEmbedResXSource() =
+    let mutable _buildEngine : IBuildEngine = null
+    let mutable _hostObject : ITaskHost = null
+    let mutable _embeddedText : ITaskItem[] = [||]
+    let mutable _generatedSource : ITaskItem[] = [||]
+    let mutable _outputPath : string = ""
+    let mutable _targetFramework : string = ""
+
+    let boilerplate = @"// <auto-generated>
+
+namespace {0}
+
+open System.Reflection
+
+module internal {1} =
+    type private C (_dummy:System.Object) = class end
+    let mutable Culture = System.Globalization.CultureInfo.CurrentUICulture
+    let ResourceManager = new System.Resources.ResourceManager(""{2}"", C(null).GetType().GetTypeInfo().Assembly)
+    let GetString(name:System.String) : System.String = ResourceManager.GetString(name, Culture)"
+
+    let boilerplateGetObject = "    let GetObject(name:System.String) : System.Object = ResourceManager.GetObject(name, Culture)"
+
+    let generateSource (resx:string) (fullModuleName:string) (generateLegacy:bool) (generateLiteral:bool) =
+        try
+            let printMessage = printfn "FSharpEmbedResXSource: %s"
+            let justFileName = Path.GetFileNameWithoutExtension(resx)
+            let namespaceName, moduleName =
```

Comment:
Please check https://github.com/Microsoft/visualfsharp/pull/3614/files#diff-f583ab0cc0cd86307b6b05f78682f65a for the hack I put in to make sure this doesn't repeatedly re-generate content, causing unexpected rebuilds on a simple repeated build like ``build net40``.  If there's a better, more declarative MSBuild way to do it without the file check comparison in the code then please let's do that.

Please also manually check that repeated ``build net40`` doesn't rebuild.

Reply:
Fixed.  Extra commits coming soon.